test-src/manageamiset_test.js

```diff
@@ -0,0 +1,144 @@
+var slugid = require('slugid');
+var assume = require('assume');
+var debug = require('debug')('test');
+var _ = require('lodash');
+var mock = require('./mock-amisets');
+var base = require('taskcluster-base');
+
+var makeAmiSet = mock.makeAmiSet;
+
+var main = require('../lib/main');
+var helper = require('./helper');
+
+describe('AMI Set api', () => {
+
+  var amiSetDefinition = makeAmiSet();
+  var amiSetChanged = _.clone(amiSetDefinition);
+  amiSetChanged.amis.push({
+    region: 'us-east-1',
+    hvm: 'ami-0123',
+    pv: 'ami-0345',
+  });
+
+  var id = amiSetDefinition.id;
+  var amis = amiSetDefinition.amis;
+
+  // testamiSetid will be used to create a second amiSet and test listAmiSets
+  var testamiSetid = 'amiset-2222';
+  var lastModified;
+
+  let amiSet;
+
+  let client;
+
+  before(async () => {
+    amiSet = await main('AmiSet', {process: 'AmiSet', profile: 'test'});
+
+    client = helper.getClient();
+  });
+
+  beforeEach(async () => {
+    await main('tableCleaner', {process: 'tableCleaner', profile: 'test'});
+  });
+
+  it('should be able to create an AMI set (idempotent)', async () => {
+    debug('### Create AMI Set');
+    await client.createAmiSet(id, amiSetDefinition);
+
+    debug('### Create AMI Set (again)');
+    await client.createAmiSet(id, amiSetDefinition);
+  });
+
+  it('should be able to update an AMI Set', async () => {
+    debug('### Load amiSet');
+    await client.createAmiSet(id, amiSetDefinition);
+
+    var amiSet = await client.getAmiSet(id);
+
+    debug('### Update amiSet');
+    try {
+      await client.updateAmiSet(id, amiSetChanged);
+    } catch (e) {
+      console.log(JSON.stringify(e));
+      throw e;
+    }
+
+    debug('### Load amiSet (again)');
+    amiSet = await client.getAmiSet(id);
+    lastModified = amiSet.lastModified;
+    assume(await amiSet).to.deeply.equal({
+      amis: [
```

Comment:
we use mock objects in a single file elsewhere to make it easier to update the format of the test objects without having to completely rewrite the test.  In this case, the objects are simple enough that I don't know if it's worth doing here, but please take a look at the `test-src/mock*.js` files just to see how they're structured.  Basically, we have a default that is returned, but you can overwrite the values you want overwritten.  Not critical here, but worth looking at.